app/workers/deliver_email_worker.rb

```diff
@@ -14,8 +14,19 @@ def self.queue_for_priority(priority)
   sidekiq_options retry: 3, queue: queue_for_priority(:low)
 
   def perform(email_id)
+    start = Time.now.utc
+
     email = Email.find(email_id)
     DeliverEmail.call(email: email)
+
+    duration = (start - Time.now.utc) * 1000.0
+
+    sleep(
+      RateSleeper.sleep_time_in_seconds(
+        execution_time_in_milliseconds: duration,
+        allowed_rate_per_second: 8
+      )
+    )
```

Comment:
You might want to only run sleep if this class returns a non zero value as (from a quick google) `sleep(0)` doesn't seem to be a no-op